docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,247 @@
+- Date: 2021-05-24
+
+Technical Story: [#8600](https://github.com/mozilla/fxa/issues/8600)
+
+## Context and Problem Statement
+
+[Mustache](https://www.npmjs.com/package/mustache), used in `fxa-auth-server` email templates, can accommodate very limited logic in its templates which has been a pain point for Firefox Accounts engineers and discussion around using a different templating system [began here](https://github.com/mozilla/fxa/issues/4627). While converting our emails to a more modern templating solution, there is an opportunity to evaluate what stack would be the most ideal for FxA emails beyond a proposed templating solution. This includes evaluating our CSS options, improving how we can preview emails in various states, and our localization tools, and the best approach for landing new templates in production.
+
+## Decisions and Considered Options
+
+### MJML
+
+[todo]
+
+### Templating
+
+- Option A - Continue to use Mustache
+- Option B - Use React server-side to generate static pages
+- Option C - Use EJS
+
+### CSS styles
+
+- Option A - Continue using ad-hoc inline styles
+- Option B - Use TailwindCSS
+- Option C - Use CSS-in-JS
+- Option D - Use a CSS pre-processor like SASS
+
+### Email previewing
+
+- Option A - Continue to use the `write-emails` command and maildev
+- Option B - Use Storybook
+- Option C - Use Mailtrap
+
+--
+
+Furthermore, there are a few **other decisions** worth noting that won’t necessarily have pros/cons lists:
+
+- How to handle the plaintext `.txt` files
+- Transitioning to Fluent over GetText for localization
+- Plans around integration involving feature flagging and the QA process
+
+## Decision Outcomes
+
+### Templating
+
+Chosen option: "Option B - Use React server-side to generate static pages", because:
+
+- The team has experience with (and likes) React
+- We're already using it in other parts of the FxA ecosystem
+- React has been tried and tested to work well with Storybook
+- React can be integrated with mjml to generate responsive emails and there is already a great [library](https://github.com/wix-incubator/mjml-react) which can help us with that purpose.
+- Components across the templates can be reused if written in React
+
+A note about tests: Since we're going with the React-based solution for our templating stack, it may make sense to use Jest to test the templates. However, we don't want a ton of time spent in test land. At the very least, we can take what we currently have and transfer/copy it, and look them over to see if anything can be improved without a high level of effort.
+
+### CSS Styles
+
+Chosen option: [todo], because:
+
+- [todo]
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and configure.
+- We have a Storybook deployment already in place and has worked great so far for a few other FxA services so hooking up the client for `fxa-auth-server` to view the modernized components might result in consistency.
```

Comment:
It has worked well, but a couple of times Storybook upgrades have borked it and it was unnoticed for a while.